docs/howtos/artifacts.rst

```diff
@@ -0,0 +1,140 @@
+    scatter_table.dataframe
+    
+The artifacts in a large composite experiment with ``flatten_results=True`` can be distinguished from
+each other using the :attr:`~.ArtifactData.experiment` and :attr:`~.ArtifactData.device_components`
+attributes.
+
+One useful pattern is to load raw or fitted data from ``curve_data`` for further data manipulation. You
+can work with the dataframe using standard pandas dataframe methods or the built-in
+:class:`.ScatterTable` methods:
+
+.. jupyter-execute::
+
+    import matplotlib.pyplot as plt
+
+    exp_type = data.artifacts(curve_data_id).experiment
+    component = data.artifacts(curve_data_id).device_components[0]
+
+    raw_data = scatter_table.filter(category="raw")
+    fitted_data = scatter_table.filter(category="fitted")
+
+    # visualize the data
+    fig, (ax1, ax2) = plt.subplots(1, 2)
+    ax1.errorbar(raw_data.x, raw_data.y, yerr=raw_data.y_err, capsize=5)
+    ax1.set_title(f"Raw data, {exp_type} experiment on {component}")
+    ax2.errorbar(fitted_data.x, fitted_data.y, yerr=fitted_data.y_err, capsize=5)
+    ax2.set_title(f"Fitted data, {exp_type} experiment on {component}")
+    plt.tight_layout()
+    plt.show()
+
+Adding artifacts
+~~~~~~~~~~~~~~~~
+
+You can add arbitrary serializable data as an artifact.
+
+.. jupyter-execute::
+
+    from qiskit_experiments.framework import ArtifactData
+
+    new_artifact = ArtifactData(name="experiment_notes", data={"content": "Testing some new ideas."})
+    data.add_artifacts(new_artifact)
+    data.artifacts("experiment_notes")
+
+.. jupyter-execute::
+
+    print(data.artifacts("experiment_notes").data)
+
+Saving and loading artifacts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note::
+    This feature is only for those who have access to the cloud service. You can 
+    check whether you do by logging into the IBM Quantum interface 
+    and seeing if you can see the `database <https://quantum.ibm.com/experiments>`__.
+
+Artifacts are saved and loaded to and from the cloud service along with the rest of the
+:class:`ExperimentData` object. Artifacts are stored as ``.zip`` files in the cloud service grouped by
+the artifact name. For example, the composite experiment above will generate two artifact files, ``fit_summary.zip`` and
+``curve_data.zip``. Each of these zipfiles will contain serialized artifact data in JSON format named
+by their unique artifact ID:
+
+.. jupyter-execute::
+    :hide-code:
+
+    print("fit_summary.zip")
+    print(f"|- {data.artifacts('fit_summary')[0].artifact_id}.json")
+    print(f"|- {data.artifacts('fit_summary')[1].artifact_id}.json")
+    print("curve_data.zip")
+    print(f"|- {data.artifacts('curve_data')[0].artifact_id}.json")
+    print(f"|- {data.artifacts('curve_data')[1].artifact_id}.json")
+    print("experiment_notes.zip")
+    print(f"|- {data.artifacts('experiment_notes').artifact_id}.json")
+
+Note that for performance reasons, the auto save feature does not apply to artifacts. You must still
```

Comment:
Yeah, I think it would be nice to change auto_save to do a single upload of figures/results/artifacts in bulk once all analysis callbacks are complete as opposed to individual calls. Artifacts is the worst to upload one by one because of the zipped files but the others have room for performance improvement too.